frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget.jsx

```diff
@@ -77,27 +90,103 @@ export default class GroupMappingsWidget extends React.Component {
     }
   };
 
-  _deleteMapping = dn => e => {
-    e.preventDefault();
+  handleShowDeleteMappingModal = (groups, dn) => {
+    this.setState({
+      showDeleteMappingModal: true,
+      showEditModal: false,
+      dnForVisibleDeleteMappingModal: dn,
+      groupIdsForVisibleDeleteMappingModal: groups,
+    });
+  };
+
+  handleHideDeleteMappingModal = () => {
+    this.setState({
+      showDeleteMappingModal: false,
+      showEditModal: true,
+      dnForVisibleDeleteMappingModal: null,
+      groupIdsForVisibleDeleteMappingModal: null,
+    });
+  };
+
+  handleConfirmDeleteMapping = (whatToDoAboutGroups, groups, dn) => {
+    this._deleteMapping(dn);
+
+    this.setState({
+      showDeleteMappingModal: false,
+      showEditModal: true,
+      dnForVisibleDeleteMappingModal: null,
+      groupIdsForVisibleDeleteMappingModal: null,
+    });
+
+    this.updateGroupsListsForCallbacksAfterDeletingMappings(
+      whatToDoAboutGroups,
+      groups,
+    );
+  };
+
+  updateGroupsListsForCallbacksAfterDeletingMappings = (
+    whatToDoAboutGroups,
+    groupIds,
+  ) => {
+    const { groups } = this.state;
+
+    if (whatToDoAboutGroups === "nothing") {
+      return;
+    }
+
+    const allGroupIdsExceptAdmin = groupIds.filter(
+      groupId => !isAdminGroup(_.find(groups, group => group.id === groupId)),
+    );
```

Comment:
Can we use `groups.find` instead of `_.find`?

Reply:
Sure, updated.